docs/operator-manual/applicationset/Generators-Plugin.md

```diff
@@ -0,0 +1,237 @@
+# Plugin Generator
```

Comment:
How should plugins support setting a requeue time? Other generators report on a per-generator basis. Should we add a new API endpoint, or let users respond with a `requeueAfter` field in the params response?

Reply:
We currently support this option like other plugins : https://github.com/argoproj/argo-cd/pull/13017/files/e73918d0a49be284a6db07f7fb8f39d4ac17c452#diff-6567471154e1860cb45c0e9ba607462fbd79d5db2292ab0ed46a981bc83c6822R33

Reply:
I was specifically thinking of a case where a plugin is receiving 5xx errors from an API and wants to back off requests. The plugin could implement its own throttling mechanism, but might be a better UX to use requeue time as a simple throttle.

Since the plugin contract is relatively difficult to break, compared to built in generators, I wonder if we should get out ahead of it.

Reply:
I don't feel super strongly about it though. A static time will likely cover most use cases.

Reply:
The idea was to keep the server part easy to develop for non advanced developers.
If a back off strategy is necessary, it should be handled by the plugin itself, or argocd as a global feature to apply to every generator.

Reply:
> If a back off strategy is necessary, it should be handled by the plugin itself

That's exactly what adding a `requeueAfter` field to the params response would do.

> or argocd as a global feature

Argo CD wouldn't have knowledge of the upstream API response, so it wouldn't know if a backoff is appropriate.

> The idea was to keep the server part easy to develop

I think we could keep server development easy. If the `requeueAfter` field isn't populated, fall back to the ApplicationSet default.

Reply:
On the other hand, if requeueing-as-backoff is an anti-pattern (CrashLoopBackoff, iirc, doesn't rely on requeue time), we should avoid it and go with static.

Reply:
I believe it's scalable, and in agreement with you, we can start statically and assess based on consumer feedback.